src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,95 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    public Object getInstance(Class<?> interfaceClass) {
+        Class<?> currentClass = findImplementation(interfaceClass);
+        Object instance = createNewInstance(currentClass);
+        initializeDependencies(instance);
+        return instance;
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceOrComponentClass) {
+        if (interfaceOrComponentClass.isInterface()) {
+            return findComponentImplementation(interfaceOrComponentClass);
+        } else if (isComponent(interfaceOrComponentClass)) {
+            return interfaceOrComponentClass;
+        } else {
+            throw new RuntimeException("Class " + interfaceOrComponentClass.getName()
+                    + " is not annotated with @Component");
+        }
+    }
+
+    private Class<?> findComponentImplementation(Class<?> interfaceClass) {
+        Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+        interfaceImplementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImplementations.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementations.put(ProductService.class, ProductServiceImpl.class);
+
+        Class<?> implementationClass = interfaceImplementations.get(interfaceClass);
+        if (implementationClass == null) {
+            throw new RuntimeException("No implementation found for interface "
+                    + interfaceClass.getName());
```

Comment:
When throwing an exception, add an informative message to it. Also, include the caught exception in the `RuntimeException` object you throw.